src/lib.rs

```diff
@@ -57,10 +57,13 @@ use core::{
     sync::atomic::Ordering,
 };
 
+#[cfg(feature = "zerocopy")]
+use zerocopy::*;
+
 // `rustfmt` keeps changing the indentation of the attributes in this macro.
 #[rustfmt::skip]
 macro_rules! define_newtype {
-    ($name:ident: $endian:ident $size_align:literal $prim:ty) => {
+    ($(#[$attr:meta])* $name:ident: $endian:ident $size_align:literal $prim:ty) => {
```

Comment:
I'd prefer the zerocopy traits be implemented separately rather than passed through as derives. This matches the impl pattern for all of the other traits and also makes it easier to consolidate all of the zerocopy impls in one place. I'd like a `zerocopy.rs` and `#[cfg(feature = "zerocopy")] mod zerocopy;` containing some macro impls. These traits are all trivially implementable anyway for rend's types.

Reply:
I assume you'd like these to be manual `unsafe impl`s rather than using zerocopy-derive?

Reply:
Update: This turns out to be impossible without relying on unstable `#[doc(hidden)]` internals.

It's doable for `IntoBytes` and `Immutable`, but `FromBytes`, `FromZeros`, `TryFromBytes`, and `KnownLayout` all require either `TryFromBytes` or `KnownLayout`, which have `#[doc(hidden)]` internals (`is_bit_valid`, `Ptr`, `DstLayout`, etc).

Are you open to using derives, or do you consider this a hard blocker?

Reply:
Derives are fine then